### DISCUSSION TITLE
New Core Web Vitals tech report

### DISCUSSION THREAD

**Discussion by sarahfossheim** [HTTPArchive/cwv-tech-report#30] [Q&A]
Some of us are currently migrating the Core Web Vitals report ([cwvtech.report](https://cwvtech.report)) off Data Studio into the HTTPArchive website. We’ll maintain feature/data parity (but with an updated interface), and are also looking at possible future additional data or ways of interacting with it.

We’d love to learn more about how you use the current report, what works well for you and what doesn’t, and what you’d like to see added/changed/fixed. 

These are the things we’d like input on, feel free to either respond in a comment, create a new issue, or contribute on Slack ⬇️

## Core Web Vitals data & insights

We want to make sure we present data that is useful, so for that we’d love to get a bit of background on what kind of (core web vitals) insights you’re after, and how you’d use those.

_Think of: What CWV related information are you after? What kind of values and metrics are you interested in? How and why do you use those? What do they tell you? Are there specific values or patterns you’re on the lookout for (eg. rising/declining values, combination of metrics), and why? What other info are you after?_

## The current report

How is the current report used, what works well, what doesn’t work well at all, and what’s missing?

_Think of: What kind of insights are you after in the cwvtech.report? Do you know beforehand or decide based on the data what you’re interested in? How do you use the report? Do you set any filters to find the information you’re after, and if so, which ones and why? Do you mainly use drilldown or compare, or both? What works well with the current tool? Is something missing in the report?_

## Sketches

These are sketches from earlier this spring. Please not that the charts have placeholder styling (so they won’t be black/white in reality) and also placeholder data, as we’re in first instance looking for feedback on the functionality, structure, filters, and chart types.

### Sketch 1:
The same graphs and filters as in the cwvtech.report, but ported over inside the HTTPArchive’s website (similar to the other reports). This is what a first feature parity without much else added could look like. Both the drilldown and comparison view can be filtered on rank, geography, origins, and be filtered further by date ranges. There's data on the different core web vitals, lighthouse scores, and page composition, all displayed over time and broken down by client. Multi-tech shows the data for the different selected technologies over time, either for mobile or desktop.

<img width="23%" alt="Landing page: showing two big card links, one for drilldown (1 tech), one for the comparison view (several tech). Underneath are two summary tables, with (mock) data on the adoption, % of origins, and good core web vitals for different technologies" src="https://github.com/HTTPArchive/cwv-tech-report/assets/10811350/055a20bd-b2b5-41bc-ba15-f9963f2bc94c"> <img width="23%" alt="Drilldown page: four cards with the latest stats on 4 different metrics (adoption, % good CWVs, lighthouse performance, page weight), then 3 line charts: % of origins with good CWVs over time, Lighthouse performance scores over time, page composition over time. Each of them has a dropdown to select different metics, and is broken down in desktop and mobile. On top of the page there's a sticky header with filters for technology, rank, geo, date" src="https://github.com/HTTPArchive/cwv-tech-report/assets/10811350/70a15cc9-d96b-4fdd-946c-da844dda9122"> <img width="23%" alt="The same drilldown report, with the CWV dropdown open. The line chart can be changed from the default % of origins with good CWVs to % of origins with good LCP, FID, CLS, FCP, TTFB, INP" src="https://github.com/HTTPArchive/cwv-tech-report/assets/10811350/c9a3d721-5920-41e7-b107-c4ddffeabc86"> <img width="23%" alt="Tech page_ Many - Unfiltered" src="https://github.com/HTTPArchive/cwv-tech-report/assets/10811350/eaa1e6e7-8e66-4695-97a1-a85eb97ac980">

### Sketch 2

A possible future version of the report, with additional metrics and graphs (such as a breakdown of the median CWVs, lighthouse audits, etc). At this point the report can be broken into several sections, for each of the big areas (CWVs, Lighthouse, etc).

<img width="23%" alt="Summary view: summary cards with the latest data on adoption, CWVs, lighthouse, page composition, and underneath a data table showing the CWVs broken down. The row headings and card headings are links." src="https://github.com/HTTPArchive/cwv-tech-report/assets/10811350/e5d8c1b3-992a-406d-ae6e-8afb2f30f1c5"> <img width="23%" alt="Usage: adoption over time, usage of the different versions, and example websites" src="https://github.com/HTTPArchive/cwv-tech-report/assets/10811350/2f53d9bc-8464-4dd0-a4e0-470d512d2f41"> <img width="23%" alt="Core Web Vitals tab: summary cards showing % of origins with good overall CWVs, FID, LCP, CLS. Then the origins with good CWVs over time line chart, which includes the median FID (latest) with a dropdown to change to different CWVs and a distribution bar chart showing the % of websites falling in different brackets of values. Finally there's a table summarising the median scores and distributions" src="https://github.com/HTTPArchive/cwv-tech-report/assets/10811350/3a1dfc19-6826-408e-916a-cfe002a03b73">  <img width="23%" alt="Opportunities tab: 4 summary cards with the latest median lighthouse scores, then underneath a table with all lighthouse audits that were flagged for origins with this tech, sorted by how often they occurred" src="https://github.com/HTTPArchive/cwv-tech-report/assets/10811350/305536e9-4ddd-4b4d-b8eb-0f7dd9ca68e1"> 

We’d love to hear what you think 😊

**Comment by kevinfarrugia** [comment]
Sketches and overall idea look very promising. 🙌

My most common use-case for using cwvtech.report is to compare different technologies using real world data. In the past I have reached out to third parties to flag a performance issue and used the cwvtech.report to illustrate the third party with similar libraries, but this would also require a manual query to extract the cause of the issue. For example, _MyFavoriteTracker_ was invalidating bfcache by using an unload listener on 100,000 sites. So it is super interesting to be able to get the lighthouse audits (sketch 2, image 4) for each technology as this would remove the requirement for running a query while making the data more accessible.

I look forward to the option to be able to filter by _rank_, to include distributions, and to get the median (or P75) value for each CWV.

I do not think it is covered in the sketches, but a small suggestion to allow adoption to be expressed as a percentage of all sites in that data set. For example, if we are looking at Wordpress, filtered by rank: 10,000, then it would be nice to easily understand that 60% (example) of sites are using Wordpress. Currently it is not possible or requires some manual computation since the total number of sites is not visible in the report.

Apart form the distribution for the last month, it would be nice if we could see the distribution over time (similar to what [treo.sh](https://treo.sh/sitespeed/web.dev?metricsMode=d) does):

![Screenshot of treo.sh showing a distribution for the current month for Largest Contentful Paint and beneath it a 100% stacked bar chart showing the distribution for each of the last 12 months](https://github.com/HTTPArchive/cwv-tech-report/assets/8075326/b7814993-ab06-4e79-91fd-8c6af0d1b31c)

Hope that helps. 😁

**Comment by Tiggerito** [comment]
Is it possible to optionally limit the reports to a specific device type?

In my distribution reports, I include a pie chart to clarify how the distribution is around the 75th percentile. Alternatively, the 75th percentile could be marked on the distribution chart (I could not work out how to do it in Looker).

   
![image](https://github.com/HTTPArchive/cwv-tech-report/assets/4947018/ccdadc60-49eb-4b53-9c39-30dfaa53da7d)

**Reply by sarahfossheim** [reply]
Thanks for the feedback! Several people have mentioned the P75 value, so we're looking into adding it 👍🏻

> Is it possible to optionally limit the reports to a specific device type?

Do you mean by mobile/desktop? If so, yes! In the current [cwvtech.report](https://lookerstudio.google.com/u/0/reporting/55bc8fad-44c2-4280-aa0b-5f3f0cd3d2be/page/M6ZPC) there already is a client dropdown in the comparison page. In the new report, both the drilldown and comparison page will be filterable by device type.

Screenshots from the client selection in the new dashboard (in development):

<img width="1436" alt="Cards with a summary of the latest data (% good CWVs, lighthouse performance, lighthouse accessibility, page weight), on top there is a 'client' dropdown in a sticky header. The text underneath the cards repeats which client this data is for." src="https://github.com/HTTPArchive/cwv-tech-report/assets/10811350/73602586-e71d-4ede-9a67-c06e349236db">

_For most components, only the selected client will be shown._

<img width="1439" alt="A timeseries chart. In the sticky header it shows the mobile client is selected. The timeseries shows both mobile and desktop." src="https://github.com/HTTPArchive/cwv-tech-report/assets/10811350/f31d2c07-4cb3-454e-a927-0532e3f004bb">

_The timeseries in the drilldown page (one tech) show both device types simultaneously._

<img width="1110" alt="Timeseries/line chart showing just mobile values, with the 'desktop' label in the legend crossed out." src="https://github.com/HTTPArchive/cwv-tech-report/assets/10811350/d5469e00-e906-4fe6-a87d-5937f907f909">

_The timeseries are filterable by client as well, directly by interacting with the chart using Highchart's built-in functionality. The legend can be used to toggle different categories on the chart on and off._

Note that all of these screenshots are from a work in progress / development version, things may change :)

**Comment by pedddro** [comment]
I think one way to improve would be to implement a way to visualize the distribution in the number of different website origins to a certain technology, that way we could [notice outliers](https://timkadlec.com/remembers/2021-01-26-what-about-ember/). Not sure if filtering outliers would be feasible to implement but it's something I think would be useful.
